drivers/max31855/max31855.c

```diff
@@ -0,0 +1,141 @@
+#include "max31855_params.h"
+
+
+void _raw_data_to_thermocouple_temperature(uint32_t raw_data, int32_t *temperature)
+{
+    assert(temperature);
+
+    int32_t sign = (raw_data >> MAX31855_THERMOCOUPLE_SIGN_SHIFT) &
+                   MAX31855_THERMOCOUPLE_SIGN_MASK ? -1 : 1;
+
+    if (sign == -1) {
+        raw_data = ~raw_data + 1;
+    }
+
+    *temperature = (raw_data >>
+                    MAX31855_THERMOCOUPLE_INTEGER_SHIFT) & MAX31855_THERMOCOUPLE_INTEGER_MASK;
+
+    /* convert to centi degC */
+    *temperature = *temperature * 100;
+
+    /* add fractional parts */
+    *temperature += (raw_data & (1 << MAX31855_THERMOCOUPLE_FRACTIONAL_HALF_SHIFT)) ? 50 : 0;
+    *temperature += (raw_data & (1 << MAX31855_THERMOCOUPLE_FRACTIONAL_QUARTER_SHIFT)) ? 25 : 0;
+
+    *temperature *= sign;
+}
+
+void _raw_data_to_internal_temperature(uint32_t raw_data, int32_t *temperature)
+{
+    assert(temperature);
+
+    int32_t sign = (raw_data >> MAX31855_INTERNAL_SIGN_SHIFT) &
+                   MAX31855_INTERNAL_SIGN_MASK ? -1 : 1;
+
+    if (sign == -1) {
+        raw_data = ~raw_data + 1;
+    }
+
+    *temperature = (raw_data >> MAX31855_INTERNAL_INTEGER_SHIFT) & MAX31855_INTERNAL_INTEGER_MASK;
+
+    /* convert to centi degC */
+    *temperature = *temperature * 100;
+
+    /* add fractional parts */
+    *temperature += (raw_data & (1 << MAX31855_INTERNAL_FRACTIONAL_HALF_SHIFT)) ? 50 : 0;
+    *temperature += (raw_data & (1 << MAX31855_INTERNAL_FRACTIONAL_QUARTER_SHIFT)) ? 25 : 0;
+    *temperature += (raw_data & (1 << MAX31855_INTERNAL_FRACTIONAL_EIGHTH_SHIFT)) ? 12 : 0;
+    *temperature += (raw_data & (1 << MAX31855_INTERNAL_FRACTIONAL_SIXTEENTH_SHIFT)) ? 6 : 0;
+
+    *temperature *= sign;
+}
+
+void _raw_data_to_fault(const uint32_t raw_data, max31855_fault_t *fault)
+{
+    assert(fault);
+
+    if (raw_data & (1 << MAX31855_FAULT_VCC_SHORT_SHIFT)) {
+        *fault = MAX31855_FAULT_VCC_SHORT;
+    }
+    else if (raw_data & (1 << MAX31855_FAULT_GND_SHORT_SHIFT)) {
+        *fault = MAX31855_FAULT_GND_SHORT;
+    }
+    else if (raw_data & (1 << MAX31855_FAULT_OPEN_CIRCUIT_SHIFT)) {
+        *fault = MAX31855_FAULT_OPEN_CIRCUIT;
+    }
+    else {
+        *fault = MAX31855_FAULT_NO_FAULT;
+    }
+}
+
+int max31855_init(max31855_t *dev, const max31855_params_t *params)
+{
+    assert(dev && params);
+    dev->params = params;
+
+    int ret = spi_init_cs(dev->params->spi, dev->params->cs_pin);
+
+    if (ret < 0) {
+        LOG_ERROR("Failed to initialize MAX31855\n");
+        return ret;
+    }
+
+    return 0;
+}
+
+void max31855_read_raw(max31855_t *dev, uint32_t *data)
+{
+    assert(dev && data);
+
+    uint8_t buffer[4] = { 0 };
+    spi_acquire(dev->params->spi, dev->params->cs_pin, SPI_MODE_0, SPI_CLK_5MHZ);
+    spi_transfer_bytes(dev->params->spi, dev->params->cs_pin, false, NULL, buffer, 4);
+    spi_release(dev->params->spi);
+
+    *data = (buffer[0] << 24) | (buffer[1] << 16) | (buffer[2] << 8) | buffer[3];
```

Comment:
What about? https://github.com/RIOT-OS/RIOT/blob/772ccb99542f13525053a6455d586667a82c2b9c/sys/include/endian.h#L63-L66

Or maybe this? https://github.com/RIOT-OS/RIOT/blob/772ccb99542f13525053a6455d586667a82c2b9c/sys/include/byteorder.h#L291